middleware/src/bitmap/soft_drawing_bitmap.adb

```diff
@@ -707,10 +704,101 @@ package body Soft_Drawing_Bitmap is
                                     D * Float (P4.Y));
          end;
       end loop;
+   end Cubic_Bezier_Points;
+
+   procedure Binomial_Coefficients
+     (Outputs     : in out NaturalArr)
+   is
+      N : constant Integer := Outputs'Length - 1;
+      F : constant Natural := Outputs'First;
+      L : constant Natural := Outputs'Last;
+      X : Natural;
+   begin
+      if N < 0 then return; end if;
+      Outputs (F) := 1;
+      Outputs (L) := 1;
+      for I in 0 .. (N / 2) - 1 loop
+         X := Outputs (F + I) * (N - I) / (I + 1);
+         Outputs (F + I + 1) := X;
+         Outputs (F + N - I - 1) := X;
+      end loop;
+   end Binomial_Coefficients;
+
+   procedure Bezier_Points
+     (InputPoints : PointsArr;
+      N           : Positive;
+      Points      : in out PointsArr)
+   is
+      Degree : constant Integer := InputPoints'Last - InputPoints'First;
+      BinomialCoeffs : NaturalArr (0 .. Degree);
+      FloatX, FloatY, T, TotalCoeff : Float;
+
+      function Power
+        (A : Float;
+         B : Natural)
+         return Float;
+
+      function Power
+        (A : Float;
+         B : Natural)
+         return Float
+      is
+         Total : Float := 1.0;
+         Count : Natural := B;
+      begin
+         while Count > 0 loop
```

Comment:
I used an exponentiation by squaring method. It would be better to use bitwise operators to check for parity and multiplying by two, but I don't know if its worth with'ing the Interfaces package